### PR TITLE
possible fix for #534

### PR DESCRIPTION
**Pull request by Skywalker-11**

### CODE REVIEW

File: mailscanner/database.php [hunk 1]
```diff
@@ -67,7 +67,12 @@ public static function connect($host = '', $username = '', $password = '', $data
      */
     public static function close()
     {
-        return self::$link->close();
+        $result = $true;
+        if (self::$link != null) {
```
Comment (endelwar) [inline mailscanner/database.php]: what about 
```php 
if (self::$link instanceof mysqli) { 
```
? (also for line 50)

File: mailscanner/database.php [hunk 2]
```diff
@@ -32,7 +32,7 @@
 class database
 {
     /** @var mysqli $link */
-    public static $link;
+    public static $link = null;
```
Comment (endelwar) [inline mailscanner/database.php]: class propriety default value is already NULL if not specified

Comment (Skywalker-11) [inline mailscanner/database.php]: done

File: mailscanner/database.php (hunk 1 above)
Comment (Skywalker-11) [inline mailscanner/database.php]: done

File: mailscanner/database.php [hunk 3]
```diff
@@ -47,7 +47,7 @@ private function __construct()
      */
     public static function connect($host = '', $username = '', $password = '', $database = '')
     {
-        if (self::$link == null || !self::$link) {
+        if (self::$link instanceof mysqli) {
```
Comment (endelwar) [inline mailscanner/database.php]: this should be negative:
`if (!self::$link instanceof mysqli) {`
`instanceof` has precedence over `!` as operator

Comment (Skywalker-11) [inline mailscanner/database.php]: done

**Pull request merged**